fix functionalization resize stride compute

bdhirsh opened the pull request:
uncovered from an OpInfo in inductor, when I turned on functionalization

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #93994
* #92857
* #94195
* __->__ #94018
* #92924

bdhirsh committed:
fix functionalization resize stride compute

[ghstack-poisoned]

ezyang reviewed (changes requested):
No, I don't understand. The old code looks right and the new code looks like you do an out of bounds strides access on the first iteration

bdhirsh committed:
Update on "fix functionalization resize stride compute"

uncovered from an OpInfo in inductor, when I turned on functionalization




[ghstack-poisoned]

bdhirsh commented:
yep you're right - somehow I convinced myself that my test passed locally. Updated, to use a convenient existing util that we have (the bug was that I should have been doing `strides[i] = strides[i+1] * sizes[i+1]` instead of `strides[i] = strides[i+1] * sizes[i]`).

bdhirsh committed:
Update on "fix functionalization resize stride compute"

uncovered from an OpInfo in inductor, when I turned on functionalization




[ghstack-poisoned]

bdhirsh committed:
Update on "fix functionalization resize stride compute"

uncovered from an OpInfo in inductor, when I turned on functionalization




[ghstack-poisoned]

bdhirsh committed:
Update on "fix functionalization resize stride compute"

uncovered from an OpInfo in inductor, when I turned on functionalization




[ghstack-poisoned]

bdhirsh committed:
Update on "fix functionalization resize stride compute"

uncovered from an OpInfo in inductor, when I turned on functionalization




[ghstack-poisoned]

bdhirsh committed:
Update on "fix functionalization resize stride compute"

uncovered from an OpInfo in inductor, when I turned on functionalization




[ghstack-poisoned]

bdhirsh committed:
Update on "fix functionalization resize stride compute"

uncovered from an OpInfo in inductor, when I turned on functionalization




[ghstack-poisoned]

bdhirsh committed:
Update on "fix functionalization resize stride compute"

uncovered from an OpInfo in inductor, when I turned on functionalization




[ghstack-poisoned]

bdhirsh committed:
Update on "fix functionalization resize stride compute"

uncovered from an OpInfo in inductor, when I turned on functionalization




[ghstack-poisoned]

bdhirsh committed:
Update on "fix functionalization resize stride compute"

uncovered from an OpInfo in inductor, when I turned on functionalization




[ghstack-poisoned]

bdhirsh committed:
Update on "fix functionalization resize stride compute"

uncovered from an OpInfo in inductor, when I turned on functionalization




[ghstack-poisoned]

bdhirsh committed:
Update on "fix functionalization resize stride compute"

uncovered from an OpInfo in inductor, when I turned on functionalization




[ghstack-poisoned]

bdhirsh committed:
Update on "fix functionalization resize stride compute"

uncovered from an OpInfo in inductor, when I turned on functionalization




[ghstack-poisoned]

bdhirsh committed:
Update on "fix functionalization resize stride compute"

uncovered from an OpInfo in inductor, when I turned on functionalization




[ghstack-poisoned]

bdhirsh committed:
Update on "fix functionalization resize stride compute"

uncovered from an OpInfo in inductor, when I turned on functionalization




[ghstack-poisoned]
